unidist/core/backends/mpi/core/worker/task_store.py

```diff
@@ -255,6 +258,7 @@ async def execute():
                             output_data_ids
                         )
                 # Monitor the task execution
+                # a blocking send is used because completion is necessary for processing to continue
```

Comment:
```suggestion
                # We use a blocking send here because we have to wait for
                # completion of the communication, which is necessary for the pipeline to continue.
```